src/libraries/System.Formats.Tar/tests/TarEntry/TarEntryGnu.Tests.cs

```diff
@@ -38,6 +38,220 @@ public void Constructor_UnsupportedEntryTypes()
+            using (TarWriter writer = new TarWriter(destination, TarEntryFormat.Gnu, leaveOpen: true))
+            {
+                writer.WriteEntry(gnuEntry); // Write DataStream exactly where the wrappedSource position was left
+            }
+
+            destination.Position = 0; // Rewind
+            using (TarReader destinationReader = new TarReader(destination, leaveOpen: false))
+            {
+                GnuTarEntry resultEntry = destinationReader.GetNextEntry() as GnuTarEntry;
+                Assert.NotNull(resultEntry);
+                using (StreamReader streamReader = new StreamReader(resultEntry.DataStream))
+                {
+                    Assert.Equal("Hello file", streamReader.ReadToEnd());
+                }
+            }
+        }
+
+        [Fact]
+        public void Constructor_ConversionFromPax_From_UnseekableTarReader()
+        {
+            using MemoryStream source = GetTarMemoryStream(CompressionMethod.Uncompressed, TestTarFormat.pax, "file");
+            using WrappedStream wrappedSource = new WrappedStream(source, canRead: true, canWrite: false, canSeek: false);
+
+            using TarReader sourceReader = new TarReader(wrappedSource, leaveOpen: true);
+            PaxTarEntry paxEntry = sourceReader.GetNextEntry(copyData: false) as PaxTarEntry;
+            GnuTarEntry gnuEntry = new GnuTarEntry(other: paxEntry); // Convert, and avoid advancing wrappedSource position
+
+            using MemoryStream destination = new MemoryStream();
+            using (TarWriter writer = new TarWriter(destination, TarEntryFormat.Gnu, leaveOpen: true))
+            {
+                writer.WriteEntry(gnuEntry); // Write DataStream exactly where the wrappedSource position was left
+            }
+
+            destination.Position = 0; // Rewind
+            using (TarReader destinationReader = new TarReader(destination, leaveOpen: false))
+            {
+                GnuTarEntry resultEntry = destinationReader.GetNextEntry() as GnuTarEntry;
+                Assert.NotNull(resultEntry);
+                using (StreamReader streamReader = new StreamReader(resultEntry.DataStream))
+                {
+                    Assert.Equal("Hello file", streamReader.ReadToEnd());
+                }
+            }
+        }
+
+        [Fact]
+        public void Constructor_ConversionV7_BackAndForth()
+        {
+            DateTimeOffset firstNow = DateTimeOffset.UtcNow - TimeSpan.FromMilliseconds(10);
+            // V7 does not support blockdev, so can't verify transfer of DeviceMajor/DeviceMinor fields
+            GnuTarEntry firstEntry = new GnuTarEntry(TarEntryType.RegularFile, "file.txt")
+            {
+                Gid = TestGid,
+                GroupName = TestGName,
+                Uid = TestUid,
+                UserName = TestUName,
+            };
+
+            Assert.True(firstEntry.AccessTime > firstNow);
+            Assert.True(firstEntry.ChangeTime > firstNow);
```

Comment:
I think this is asserting that the clock is monotonic (never goes backwards), which isn't a good assumption.  The test should instead just assign an ATime and CTime value to something far in the past (e.g. yesterday's date as a fixed value... that's "far" as far as clock drift/NTP is concerned, and gets farther every day).

Then, when coming back from GNU -> V7 -> GNU see that (a) the new times are not equal to the old, (b) they're UTC (Offset == TimeSpan.Zero), and (c) that they're some value close to "now" (e.g. `Assert.Equal(firstNow, secondEntry.AccessTime, TimeSpan.FromMinutes(5))` for the built-in comparator with 5 minutes tolerance -- if that doesn't compile, then it might only exist for DateTime values, but you can just use firstNow.UtcDateTime and secondEntry.AccessTime.UtcDateTime after verifying you got UTC)